src/states/reducer.js

```diff
@@ -0,0 +1,85 @@
+import CONSTANTS from './constants';
+
+const ACTIONS = {
+    LOGIN: 'LOGIN',
+    LOGOUT: 'LOGOUT',
+    CHANGE_SELECTED_THEME: 'CHANGE_SELECTED_THEME',
+    CHANGE_RESPONSE: 'CHANGE_RESPONSE',
+    CHANGE_SEARCH_VALUE: 'CHANGE_SEARCH_VALUE',
+    CHANGE_SELECTED_VIDEO: 'CHANGE_SELECTED_VIDEO',
+    ADD_TO_FAVORITES: 'ADD_TO_FAVORITES',
+    REMOVE_FROM_FAVORITES: 'REMOVE_FROM_FAVORITES',
+};
+
+function handleFavorite(isAddOperation, video) {
+    const favorites = new Map(JSON.parse(localStorage.getItem(CONSTANTS.FAVORITE_KEY)));
+
+    if (video && video.id.videoId) {
+        if (isAddOperation) {
+            favorites.set(video.id.videoId, video);
+        } else {
+            favorites.delete(video.id.videoId);
+        }
+
+        localStorage.setItem(
+            CONSTANTS.FAVORITE_KEY,
+            JSON.stringify(Array.from(favorites.entries()))
+        );
+    } else {
+        console.log(
```

Comment:
In a real-life scenario it would make sense to also handle errors in the UI so the user is aware of it!